### PR TITLE
Ext Angular: Incremental PR

### PR DESCRIPTION
**Pull request by aniket91**
This PR contains -
1. FIxed Heterogeneous tree example in Kichensink
2. Added more Sencha tests  (todo: https://github.com/sencha/ext-angular/blob/31d8c05e7c198dde4c54dd2f056eb90f5057db6a/packages/ext-angular-tests/src/app/test/TODO)
3. Updated boiler plate code for angular modern to be same as react.
4. Rest of api docs for angular

### CODE REVIEW

**Comment by branflake2267** [comment]
Could you add `@example packages=[angular]` for the @examples? Unless we want to rewire the docs system.

File: packages/ext-angular-modern-boilerplate/src/app/app.component.html [hunk 1]
```diff
@@ -0,0 +1,22 @@
+<container #item  fullscreen="true" layout="fit">
+    <titlebar #item  [title]="this.title" docked="top">
+            <button #item  align="left" *ngIf="isPhone" iconCls="x-fa fa-bars" [handler]="this.toggleAppMenu" ripple="false"></button>
+    </titlebar>
+
+        <sheet #item  *ngIf="isPhone" [displayed]="showAppMenu" side="left" (hide)="this.onHideAppMenu()">
+            <panel #item  scrollable="true" title="ExtReact Boilerplate">
```
Comment (branflake2267) [inline packages/ext-angular-modern-boilerplate/src/app/app.component.html]: React?

File: packages/ext-angular-tests/src/app/test/ReplaceNodeWithMarkup/ReplaceNodeWithMarkup.html [hunk 2]
```diff
@@ -0,0 +1,14 @@
+<container #item>
+        <container #item>
+            <div #item>
+                This tests that ExtAngular's patching of ReactComponentEnvironment.replaceNodeWithMarkup correctly adds and
```
Comment (branflake2267) [inline packages/ext-angular-tests/src/app/test/ReplaceNodeWithMarkup/ReplaceNodeWithMarkup.html]: React?

File: packages/ext-angular-tests/src/app/test/TODO [hunk 3]
```diff
@@ -0,0 +1,24 @@
+>>>Remaining components<<<
```
Comment (branflake2267) [inline packages/ext-angular-tests/src/app/test/TODO]: Do you want this included? It also has React in it?

File: packages/ext-angular-tests/src/app/test/Template/TemplateGrid/TemplateGrid.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+<container #item layout="vbox">
+    <container #item>
+        <div #item>
+            This tests that we can render React elements in grid cells using the tpl config.
```
Comment (branflake2267) [inline packages/ext-angular-tests/src/app/test/Template/TemplateGrid/TemplateGrid.html]: React?

File: packages/ext-angular-tests/src/app/test/Template/TemplateList/TemplateList.html [hunk 5]
```diff
@@ -0,0 +1,12 @@
+<container #item>
+    <container #item>
+        <div #item>This tests that we can render React elements including ExtReact components in list rows.
```
Comment (branflake2267) [inline packages/ext-angular-tests/src/app/test/Template/TemplateList/TemplateList.html]: React?

File: packages/ext-angular-tests/test/All_Tests/CreateChild.js [hunk 6]
```diff
@@ -4,7 +4,7 @@ describe("CreateChild", function () {
         ST.button('#insert').click();
         ST.component('#inserted').and(function(inserted) {
             expect(inserted.getParent().indexOf(inserted)).toBe(0);
-        })
+        });
     });
 
     it("should insert an ExtReact component in the middle", function () {
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/CreateChild.js]: React?

File: packages/ext-angular-tests/test/All_Tests/HTML.js [hunk 7]
```diff
@@ -3,9 +3,9 @@ describe("HTML", () => {
         ST.navigate('#/HTMLInExtReactComponent')
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/HTML.js]: React?

File: packages/ext-angular-tests/test/All_Tests/Lifecycle.js [hunk 8]
```diff
@@ -3,7 +3,7 @@ describe("Lifecycle", () => {
         it('should be called when a parent ExtReact component is removed', () => {
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/Lifecycle.js]: React?

File: packages/ext-angular-tests/test/All_Tests/SenchaTestHooks.js [hunk 9]
```diff
@@ -4,12 +4,12 @@ describe('Sencha Test Hooks', () => {
     it('should include $ExtReactComponentName on each rendered component', () => {
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/SenchaTestHooks.js]: React?

File: packages/ext-angular-tests/test/All_Tests/Template.js [hunk 10]
```diff
@@ -2,7 +2,7 @@ describe('Template', () => {
     describe("Column.tpl", () => {
         it('should render React elements returned by cell tpl', () => {
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/Template.js]: React?

File: packages/ext-angular-tests/test/All_Tests/Template.js [hunk 11]
```diff
@@ -11,19 +11,23 @@ describe('Template', () => {
     describe("List.itemTpl", () => {
         it('should render React elements returned by itemTpl', () => {
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/Template.js]: React?

File: packages/ext-angular-tests/test/All_Tests/Template.js [hunk 12]
```diff
@@ -11,19 +11,23 @@ describe('Template', () => {
...
+            
+            ST.element('@container-div')
+            .child('button')
+            .visible();
+            
         })
     });
 
     describe("unmount", () => {
         it('should unmount components when the parent ExtReact component is destroyed', () => {
```
Comment (branflake2267) [inline packages/ext-angular-tests/test/All_Tests/Template.js]: React?

File: api-docs/modern/modern/src/dataview/ChipView.js [hunk 13]
```diff
@@ -0,0 +1,142 @@
+/**
+ * A specialized `Ext.dataview.DataView` to display items based upon the structure of
+ * the `Ext.Chip` component with the configurations of the "chips" drawn from named
+ * fields in the DataView's records.
+ *
+ * This handles selection and deletion of chips.
+ *     @example
```
Comment (branflake2267) [inline api-docs/modern/modern/src/dataview/ChipView.js]: For the @examples can you change them to: `@example packages=[extangular]` and fix the spacing?

File: api-docs/modern/modern/src/colorpicker/Button.js [hunk 14]
```diff
@@ -0,0 +1,41 @@
+/**
+ * A simple color swatch that can be clicked to bring up the color selector.
+ *
+ * The selected color is configurable via {@link #value} and
+ * The Format is configurable via {@link #format}.
+ *
+ *            @example
```
Comment (branflake2267) [inline api-docs/modern/modern/src/colorpicker/Button.js]: For the @examples can you change them to: `@example packages=[extangular]`?

File: api-docs/ux/gauge/needle/Diamond.js [hunk 15]
```diff
@@ -1,50 +1,46 @@
...
- *                  </ExtReact>
- *              )
- *          }
- *      }
- */
\ No newline at end of file
+ *      import { Component } from '@angular/core'
+ *        declare var Ext: any;
+ *
+ *        @Component({
```
Comment (branflake2267) [inline api-docs/ux/gauge/needle/Diamond.js]: Would you like to add an @example?

File: packages/ext-angular-modern-boilerplate/src/app/About/about.component.ts [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+
+@Component({
+  selector: 'app-about',
+  templateUrl: './about.component.html',
+  styles: [``]
+})
+export class AboutComponent implements OnInit {
+
+  constructor() { }
+
```
Comment (branflake2267) [inline packages/ext-angular-modern-boilerplate/src/app/About/about.component.ts]: Is all this space intentional? Could it be removed?

File: packages/ext-angular-modern-boilerplate/src/app/Home/home.component.ts [hunk 17]
```diff
@@ -0,0 +1,59 @@
...
+declare var Ext: any;
+
+@Component({
+  selector: 'app-home',
+  templateUrl: './home.component.html',
+  styles: [``]
+})
+export class HomeComponent implements OnInit {
+
+  constructor() { }
```
Comment (branflake2267) [inline packages/ext-angular-modern-boilerplate/src/app/Home/home.component.ts]: Could you remove the new lines and properly indent the class members?

File: packages/ext-angular-modern-boilerplate/src/app/app.component.ts [hunk 18]
```diff
@@ -1,35 +1,56 @@
...
+navStore = {
+	root: {
+			children: [
+					{ id: '/', text: 'Home', iconCls: 'x-fa fa-home', leaf: true },
+					{ id: '/about', text: 'About', iconCls: 'x-fa fa-info', leaf: true },
+			]
+	}
+};
+
+responsiveConfig = {
```
Comment (branflake2267) [inline packages/ext-angular-modern-boilerplate/src/app/app.component.ts]: The class member indentation doesn't look right here?

File: packages/ext-angular-tests/src/app/test/Lifecycle/MyComponent/MyComponent.ts [hunk 19]
```diff
@@ -0,0 +1,21 @@
+import { Component, OnInit, ViewChild, ElementRef } from '@angular/core';
+
+declare var Ext: any;
+
+@Component({
+  selector: 'my-component',
+  templateUrl: './MyComponent.html',
+})
+
```
Comment (branflake2267) [inline packages/ext-angular-tests/src/app/test/Lifecycle/MyComponent/MyComponent.ts]: Could you remove all new lines after the decorator and before the class? 

Here is a good example: https://angular.io/guide/architecture-components#component-metadata

File: packages/ext-angular-modern-boilerplate/src/app/app.component.html (hunk 1 above)
Comment (aniket91) [reply packages/ext-angular-modern-boilerplate/src/app/app.component.html]: Fixed.

File: packages/ext-angular-modern-boilerplate/src/app/About/about.component.ts (hunk 16 above)
Comment (aniket91) [reply packages/ext-angular-modern-boilerplate/src/app/About/about.component.ts]: Fixed.

File: packages/ext-angular-modern-boilerplate/src/app/Home/home.component.ts (hunk 17 above)
Comment (aniket91) [reply packages/ext-angular-modern-boilerplate/src/app/Home/home.component.ts]: Fixed.

File: packages/ext-angular-modern-boilerplate/src/app/app.component.ts (hunk 18 above)
Comment (aniket91) [reply packages/ext-angular-modern-boilerplate/src/app/app.component.ts]: Fixed.

File: packages/ext-angular-tests/src/app/test/Lifecycle/MyComponent/MyComponent.ts (hunk 19 above)
Comment (aniket91) [reply packages/ext-angular-tests/src/app/test/Lifecycle/MyComponent/MyComponent.ts]: Fixed.

File: packages/ext-angular-tests/src/app/test/ReplaceNodeWithMarkup/ReplaceNodeWithMarkup.html (hunk 2 above)
Comment (aniket91) [reply packages/ext-angular-tests/src/app/test/ReplaceNodeWithMarkup/ReplaceNodeWithMarkup.html]: Fixed.

File: packages/ext-angular-tests/src/app/test/TODO (hunk 3 above)
Comment (aniket91) [reply packages/ext-angular-tests/src/app/test/TODO]: @branflake2267  Yes, these are known issues that we are facing with respect to sencha tests.

File: packages/ext-angular-tests/src/app/test/Template/TemplateGrid/TemplateGrid.html (hunk 4 above)
Comment (aniket91) [reply packages/ext-angular-tests/src/app/test/Template/TemplateGrid/TemplateGrid.html]: Fixed.

File: packages/ext-angular-tests/src/app/test/Template/TemplateList/TemplateList.html (hunk 5 above)
Comment (aniket91) [reply packages/ext-angular-tests/src/app/test/Template/TemplateList/TemplateList.html]: Fixed.

File: packages/ext-angular-tests/test/All_Tests/CreateChild.js (hunk 6 above)
Comment (aniket91) [reply packages/ext-angular-tests/test/All_Tests/CreateChild.js]: FIxed.

File: packages/ext-angular-tests/test/All_Tests/HTML.js (hunk 7 above)
Comment (aniket91) [reply packages/ext-angular-tests/test/All_Tests/HTML.js]: Fixed.

File: packages/ext-angular-tests/test/All_Tests/SenchaTestHooks.js (hunk 9 above)
Comment (aniket91) [reply packages/ext-angular-tests/test/All_Tests/SenchaTestHooks.js]: Fixed.

File: packages/ext-angular-tests/test/All_Tests/Template.js (hunk 10 above)
Comment (aniket91) [reply packages/ext-angular-tests/test/All_Tests/Template.js]: Fixed.

File: packages/ext-angular-tests/test/All_Tests/Template.js (hunk 11 above)
Comment (aniket91) [reply packages/ext-angular-tests/test/All_Tests/Template.js]: Fixed.

File: packages/ext-angular-tests/test/All_Tests/Template.js (hunk 12 above)
Comment (aniket91) [reply packages/ext-angular-tests/test/All_Tests/Template.js]: Fixed.

File: api-docs/ux/gauge/needle/Diamond.js (hunk 15 above)
Comment (aniket91) [reply api-docs/ux/gauge/needle/Diamond.js]: Fixed.

**Pull request merged**